### ISSUE TITLE
Close PR Trigger should be removed in some cases

### ISSUE THREAD

**Issue by yp05327** [go-gitea/gitea#23222]
### Description

I used both `push` and `pull_request` in the workflow config yaml file.
``` yaml
on:
  - push
  - pull_request
```

If you merge a PR, you will get two action tasks:
![image](https://user-images.githubusercontent.com/18380374/222160295-8f06bbb8-9f3f-4c61-ba94-fd16f7540ba7.png)

In `action_run` table, the record is as following:
![image](https://user-images.githubusercontent.com/18380374/222161751-c2dad7d8-1451-4dce-af48-c414b2ea597e.png)

ref name is `pr.MergedCommitID`, and the event's action is `closed`
https://github.com/go-gitea/gitea/blob/7a5af25592003ddc3017fcd7b822a3e02fc40ef6/services/actions/notifier.go#L337

It seems that Close PR Event will be triggered, when you merge a PR.   
I don't think we need to trigger Close PR Event, as if a PR is closed without merge, this event will also be triggered:
Create a PR and just close it:
![image](https://user-images.githubusercontent.com/18380374/222165766-5d54fd2a-b0e4-43a4-ab1d-f05ee8ec417b.png)
An action task will start to run:
![image](https://user-images.githubusercontent.com/18380374/222165674-a31eeefc-a989-4337-b33f-37ca7373b622.png)
And the action task name will be the latest commit name:
![image](https://user-images.githubusercontent.com/18380374/222167718-a9a38e0a-4e26-45da-99e6-a4182d8b18dd.png)

If we removed Close PR Trigger, when a PR is merged, the only Push Trigger will work.

### Gitea Version

latest

### Can you reproduce the bug on the Gitea demo site?

No

### How are you running Gitea?

build

### Database

None

**Comment by wolfogre** [comment]
TBH, I noticed this bug early on, but didn't know how to fix it.

It's not a good idea to remove the close PR trigger. Think about this, what if I want a workflow to add a comment to a PR automatically when the PR has been closed?

**Comment by lunny** [comment]
So the default events for `pull_request` is only `create` and `sync`?

**Comment by yp05327** [comment]
@wolfogre 
You are right. I will change the title.

I found this document:
https://docs.github.com/ja/actions/using-workflows/events-that-trigger-workflows#running-your-pull_request-workflow-based-on-the-head-or-base-branch-of-a-pull-request
It seems that we can set activity types in github.

Then I changed workflow yaml file:
``` yaml
name: Gitea Actions Demo
run-name: ${{ github.actor }} is testing out Gitea Actions 🚀
on:
  pull_request:
    types: [opened]

jobs:
  Explore-Gitea-Actions:
    runs-on: ubuntu-latest
    steps:
      - run: echo "🎉 The job was automatically triggered by a ${{ github.event_name }} event."
      - run: echo "🐧 This job is now running on a ${{ runner.os }} server hosted by Gitea!"
      - run: echo "🔎 The name   of your branch is ${{ github.ref }} and your repository is ${{ github.repository }}."
      - name: Check out repository code
        uses: actions/checkout@v3
      - run: echo "💡 The ${{ github.repository }} repository has been cloned to the runner."
      - run: echo "🖥️ The workflow is now ready to test your code on the runner."
      - name: List files in the repository
        run: |
          ls ${{ github.workspace }}          
      - run: echo "🍏 This job's status is ${{ job.status }}."
```
It seems that act is not support this:
![image](https://user-images.githubusercontent.com/18380374/222330687-4f948948-394d-4a44-a41a-1132f39b45e2.png)

`RawOn` is defined as a `yaml.Node`
![image](https://user-images.githubusercontent.com/18380374/222333780-bf7e35aa-2801-4eb5-b941-82c006486915.png)

**Comment by lunny** [comment]
This maybe fixed by #23037

**Comment by yp05327** [comment]
> It's not a good idea to remove the close PR trigger. Think about this, what if I want a workflow to add a comment to a PR automatically when the PR has been closed?

I think it is possible to remove close PR trigger in some cases as default, such as close PRs, merge PRs(which will trigger close PR event), as if users want to trigger these close PR events, they can add `types` to customize their needs.

**Comment by yp05327** [comment]
Fixed by #23037
